wrappers/src/object_cs.cpp

```diff
@@ -219,175 +219,183 @@ extern "C" {
+            if (object_ptr.row().get_table()->find_first_string(column_ndx, str) != not_found) {
                 throw SetDuplicatePrimaryKeyValueException(
-                                                           object_ptr->row().get_table()->get_name(),
-                                                           object_ptr->row().get_table()->get_column_name(column_ndx),
+                                                           object_ptr.row().get_table()->get_name(),
+                                                           object_ptr.row().get_table()->get_column_name(column_ndx),
                                                            str.to_string()
                                                            );
             }
-            object_ptr->row().set_string_unique(column_ndx, str);
+            object_ptr.row().set_string_unique(column_ndx, str);
         });
     }
     
-    REALM_EXPORT void object_set_binary(const Object* object_ptr, size_t property_ndx, char* value, size_t value_len, NativeException::Marshallable& ex)
+    REALM_EXPORT void object_set_binary(const Object& object_ptr, size_t property_ndx, char* value, size_t value_len, NativeException::Marshallable& ex)
     {
         return handle_errors(ex, [&]() {
             verify_can_set(object_ptr);
             
             const size_t column_ndx = get_column_index(object_ptr, property_ndx);
-            object_ptr->row().set_binary(column_ndx, BinaryData(value, value_len));
+            object_ptr.row().set_binary(column_ndx, BinaryData(value, value_len));
         });
     }
     
-    REALM_EXPORT void object_set_timestamp_ticks(const Object* object_ptr, size_t property_ndx, int64_t value, NativeException::Marshallable& ex)
+    REALM_EXPORT void object_set_timestamp_ticks(const Object& object_ptr, size_t property_ndx, int64_t value, NativeException::Marshallable& ex)
     {
         return handle_errors(ex, [&]() {
             verify_can_set(object_ptr);
             
             const size_t column_ndx = get_column_index(object_ptr, property_ndx);
-            object_ptr->row().set_timestamp(column_ndx, from_ticks(value));
+            object_ptr.row().set_timestamp(column_ndx, from_ticks(value));
         });
     }
     
-    REALM_EXPORT void object_remove_row(Object* object_ptr, NativeException::Marshallable& ex)
+    REALM_EXPORT void object_remove_row(const Object& object_ptr, NativeException::Marshallable& ex)
     {
         handle_errors(ex, [&]() {
             verify_can_set(object_ptr);
             
-            object_ptr->row().get_table()->move_last_over(object_ptr -> row().get_index());
+            object_ptr.row().get_table()->move_last_over(object_ptr.row().get_index());
+        });
+    }
+    
+    REALM_EXPORT size_t object_equals_object(const Object& object_ptr, const Object& other_object_ptr, NativeException::Marshallable& ex)
```

Comment:
This should return `bool`. See comment on the P/Invoke declaration for this function.